cpp/src/quantiles/quantiles.cu

```diff
@@ -30,68 +30,31 @@
 
 namespace{ // anonymous
 
-  struct QuantileIndex {
-    gdf_size_type lower_bound;
-    gdf_size_type upper_bound;
-    gdf_size_type nearest;
-    double fraction;
-
-    QuantileIndex(gdf_size_type length, double quant)
-    {
-      // clamp quant value.
-      // Todo: use std::clamp if c++17 is supported.
-      quant = std::min(std::max(quant, 0.0), 1.0);
-
-      // since gdf_size_type is int32_t, there is no underflow/overflow
-      double val = quant*(length -1);
-      lower_bound = std::floor(val);
-      upper_bound = static_cast<size_t>(std::ceil(val));
-      nearest = static_cast<size_t>(std::nearbyint(val));
-      fraction = val - lower_bound;
-    }
-  };
-
-
-  template<typename T>
-  void singleMemcpy(T& res, T* input, cudaStream_t stream)
-  {
-    (void)stream;
-    //TODO: async with streams?
-    CUDA_TRY( cudaMemcpy(&res, input, sizeof(T), cudaMemcpyDeviceToHost) );
-  }
-
   // compute quantile value as `result` from `quant` value by `interpolation` method
   template<typename T, typename RetT>
   gdf_error select_quantile(T* devarr,
                           gdf_size_type n,
                           double quant, 
-                          gdf_quantile_method interpolation,
+                          cudf::quantile_method interpolation,
                           RetT& result,
                           bool flag_sorted,
                           cudaStream_t stream)
   {
-    std::vector<T> hvalue(2);
-
-    if( n < 2 )
-    {
-      singleMemcpy(hvalue[0], devarr, stream);
-      result = static_cast<RetT>( hvalue[0] );
-      return GDF_SUCCESS;
-    }
+    T hvalue;
 
     if( quant >= 1.0 && !flag_sorted )
     {
       T* d_res = thrust::max_element(rmm::exec_policy(stream)->on(stream), devarr, devarr+n);
-      singleMemcpy(hvalue[0], d_res, stream);
-      result = static_cast<RetT>( hvalue[0] );
+      cudf::detail::singleMemcpy(hvalue, d_res);

```

Comment:
I thought you removed this function. If it *must* stay, please rename it. (underscores and lower case, not CamelCase). I think "single memcpy" is unclear. It's `copy_single_element`.  

But I think it's not necessary at all now that you only have two calls to it left.

Reply:
Somehow my environment doesn't rebuild sources that depend on headers i touch. This should've been caught.